cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -373,6 +373,26 @@ def create_service_configs(kube_control):
     remove_state('authentication.setup')
 
 
+@when('kube-control.departed')
+@when('leadership.is_leader')
+def flush_auth_for_departed(kube_control):
+    ''' Unit has left the cluster and needs to have its authentication
+    tokens removed from the token registry '''
+    token_auth_file = '/root/cdk/known_tokens.csv'
+    departing_unit = kube_control.flush_departed()
+    known_tokens = open(token_auth_file, 'r').readlines()
+    for line in known_tokens:
+        if departing_unit in line:
+            hookenv.log('Found unit {} in token auth. Removing auth token.')
+            known_tokens.remove(line)
```

Comment:
I think removing from `known_tokens` while we're iterating over it could be bad, see https://stackoverflow.com/a/1207427